src/pages/App/App.js

```diff
@@ -93,32 +111,23 @@ export default function App({ song }) {
                         <audio
                             id="audio"
                             onEnded={onSongEnd}
-                            onLoadedMetadata={e =>
-                                setDuration(e.currentTarget.duration)
-                            }
+                            onLoadedMetadata={e => dispatch(setDuration(parseInt(e.currentTarget.duration)))}
                             onPlay={() => setIsPlaying(true)}
-                            onTimeUpdate={e =>
-                                !isSongEnded &&
-                                parseInt(e.target.currentTime) !==
-                                    parseInt(currentTime) &&
-                                setCurrentTime(e.target.currentTime)
-                            }
+                            onTimeUpdate={onTimeUpdateHandler}
                         ></audio>
                     </div>
                     <div className={classes.bar}>
                         <PlayerBar
                             currentTime={currentTime}
                             volume={volume}
                             onVolumeChange={e => setVolume(e.target.value)}
-                            onPlayPress={() => setPlayPressed(!playPressed)}
-                            playPressed={playPressed}
+                            onPlayPress={() => playPressedHandler(!isPlayPressed)}
+                            playPressed={isPlayPressed}
                             isPlaying={isPlaying}
-                            uploadedSong={song}
```

Comment:
As for comment above, 
You can go to PlayerBar.is and there change the reference of { uploaded song } prop to redux store state, and on the way delete few other props as { playPressed, onPlayPress} etc..

Reply:
True! Already did some cleaning in PlayerBar.js and Visualizer.js and App.js